test/FingerprintApiTest.php

```diff
@@ -280,4 +343,173 @@ public function testGetBrokenFormatEvent()
+        } catch (ApiException $e) {
+            $this->assertEquals("VisitorNotFound", $e->getErrorDetails()->getError()->getCode());
+
+            throw $e;
+        }
+    }
+
+    public function testDeleteVisitorData429Error()
+    {
+        $this->mockHandler->reset();
+        $this->mockHandler->append($this->getMockResponse(self::MOCK_VISITOR_ID_429_ERROR, "DELETE"));
+
+        $this->expectException(ApiException::class);
+        $this->expectExceptionCode(429);
+
+        $this->fingerprint_api->deleteVisitorData(self::MOCK_VISITOR_ID_429_ERROR);
+    }
+
+    public function testUpdateEvent400Error()
+    {
+        $this->mockHandler->reset();
+        $this->mockHandler->append($this->getMockResponse(self::MOCK_EVENT_ID_400_ERROR, "PUT"));
+
+        $this->expectException(ApiException::class);
+        $this->expectExceptionCode(400);
+
+        try {
+            $this->fingerprint_api->updateEvent(new EventUpdateRequest([]), self::MOCK_EVENT_ID_400_ERROR);
+        } catch (ApiException $e) {
+            $this->assertEquals("RequestCannotBeParsed", $e->getErrorDetails()->getError()->getCode());
+            throw $e;
+        }
+    }
+
+    public function testUpdateEvent403Error()
+    {
+        $this->mockHandler->reset();
+        $this->mockHandler->append($this->getMockResponse(self::MOCK_EVENT_ID_403_ERROR, "PUT"));
+
+
+        $this->expectException(ApiException::class);
+        $this->expectExceptionCode(403);
+
+        try {
+            $this->fingerprint_api->updateEvent(new EventUpdateRequest([]), self::MOCK_EVENT_ID_403_ERROR);
+        } catch (ApiException $e) {
+            $this->assertEquals("TokenRequired", $e->getErrorDetails()->getError()->getCode());
+            throw $e;
+        }
+    }
+
+    public function testUpdateEvent404Error()
+    {
+        $this->mockHandler->reset();
+        $this->mockHandler->append($this->getMockResponse(self::MOCK_EVENT_ID_404_ERROR, "PUT"));
+
+        $this->expectException(ApiException::class);
+        $this->expectExceptionCode(404);
+
+        try {
+            $this->fingerprint_api->updateEvent(new EventUpdateRequest([]), self::MOCK_EVENT_ID_404_ERROR);
+        } catch (ApiException $e) {
+
+            $this->assertEquals("RequestNotFound", $e->getErrorDetails()->getError()->getCode());
+            throw $e;
+        }
+    }
+
+    public function testUpdateEvent409Error()
+    {
+        $this->mockHandler->reset();
+        $this->mockHandler->append($this->getMockResponse(self::MOCK_EVENT_ID_409_ERROR, "PUT"));
+
+        $this->expectException(ApiException::class);
+        $this->expectExceptionCode(409);
+
+        try {
+            $this->fingerprint_api->updateEvent(new EventUpdateRequest([]), self::MOCK_EVENT_ID_409_ERROR);
+        } catch (ApiException $e) {
+            $this->assertEquals("StateNotReady", $e->getErrorDetails()->getError()->getCode());
```

Comment:
Nice, I like it!
Could fix other similar tests too?

Reply:
Last commit should fix this